lib/Sema/TypeCheckProtocolInference.cpp

```diff
@@ -200,7 +200,7 @@ AssociatedTypeInference::inferTypeWitnessesViaValueWitnesses(
     // The extension may not have a generic signature set up yet, as a
     // recursion breaker, in which case we can't yet confidently reject its
     // witnesses.
-    if (!extension->getGenericSignature())
+    if (!extension->hasComputedGenericSignature())
```

Comment:
What happens if you just remove this?

Reply:
Asking for the generic signature kicks off type resolution which kicks off protocol witness  resolution which kicks off protocol inference which kicks off this request again. 😢

Reply:
Ah, something to untangle later then. Can you file a radar?